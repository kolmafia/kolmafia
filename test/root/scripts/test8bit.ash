location VANYAS_CASTLE = $location[Vanya's Castle];
location MEGALO_CITY = $location[Megalo-City];
location HEROS_FIELD = $location[Hero's Field];
location FUNGUS_PLAINS = $location[The Fungus Plains];

void loop8(location loc, string color)
{
    int base = 0;
    boolean bonus = false;
    switch (loc) {
    case VANYAS_CASTLE:
	base = 300;
	bonus = color == "black";
	break;
    case MEGALO_CITY:
	base = 300;
	bonus = color == "blue";
	break;
    case HEROS_FIELD:
	base = 100;
	bonus = color == "green";
	break;
    case FUNGUS_PLAINS:
	base = 150;
	bonus = color == "red";
	break;
    }
    print(loc + " base=" + base + " color=" + color);
    print("--------------");

    // Modifier value
    int minX = 0;
    int maxX = base + 300;

    // Points
    int minY = bonus ? 100 : 50;
    int maxY = bonus ? 400 : 200;

    // So we can detect when Y value (points) increases
    int curX = minx;
    int curY = miny;

    void print_range(int low, int high, int y)
    {
	string lowx = low == 0 ? "000" : to_string(low);
	string highx = to_string(high);
	print(lowx + "-" + highx + " -> " + y);
    }

    for (int x = minX; x <= maxX; ++x) {
	int points = eight_bit_points(loc, color, x);
	if (points < curY) abort(x + " -> " + points);
	if (points > curY) {
	    print_range(curX, x-1, curY);
	    curX = x;
	    curY = points;
	}
    }
    if (curX < maxX) {
	print_range(curX, maxX, curY);
    }
    print();
}

void main()
{
    void loop8(location loc, string... colors)
    {
	foreach i, color in colors {
	    loop8(loc, color);
	}
    }

    loop8(VANYAS_CASTLE, "white", "black");
    loop8(MEGALO_CITY, "white", "blue");
    loop8(HEROS_FIELD, "white", "green");
    loop8(FUNGUS_PLAINS, "white", "red");
}
