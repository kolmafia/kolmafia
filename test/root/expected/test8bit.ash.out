Vanya's Castle base=300 color=white
--------------
0-9 -> 50
10-29 -> 60
30-49 -> 70
50-69 -> 80
70-89 -> 90
90-109 -> 100
110-129 -> 110
130-149 -> 120
150-169 -> 130
170-189 -> 140
190-209 -> 150
210-229 -> 160
230-249 -> 170
250-269 -> 180
270-289 -> 190
290-300 -> 200

Vanya's Castle base=300 color=black
--------------
0-4 -> 100
5-14 -> 110
15-24 -> 120
25-34 -> 130
35-44 -> 140
45-54 -> 150
55-64 -> 160
65-74 -> 170
75-84 -> 180
85-94 -> 190
95-104 -> 200
105-114 -> 210
115-124 -> 220
125-134 -> 230
135-144 -> 240
145-154 -> 250
155-164 -> 260
165-174 -> 270
175-184 -> 280
185-194 -> 290
195-204 -> 300
205-214 -> 310
215-224 -> 320
225-234 -> 330
235-244 -> 340
245-254 -> 350
255-264 -> 360
265-274 -> 370
275-284 -> 380
285-294 -> 390
295-300 -> 400

Megalo-City base=300 color=white
--------------
0-9 -> 50
10-29 -> 60
30-49 -> 70
50-69 -> 80
70-89 -> 90
90-109 -> 100
110-129 -> 110
130-149 -> 120
150-169 -> 130
170-189 -> 140
190-209 -> 150
210-229 -> 160
230-249 -> 170
250-269 -> 180
270-289 -> 190
290-300 -> 200

Megalo-City base=300 color=blue
--------------
0-4 -> 100
5-14 -> 110
15-24 -> 120
25-34 -> 130
35-44 -> 140
45-54 -> 150
55-64 -> 160
65-74 -> 170
75-84 -> 180
85-94 -> 190
95-104 -> 200
105-114 -> 210
115-124 -> 220
125-134 -> 230
135-144 -> 240
145-154 -> 250
155-164 -> 260
165-174 -> 270
175-184 -> 280
185-194 -> 290
195-204 -> 300
205-214 -> 310
215-224 -> 320
225-234 -> 330
235-244 -> 340
245-254 -> 350
255-264 -> 360
265-274 -> 370
275-284 -> 380
285-294 -> 390
295-300 -> 400

Hero's Field base=100 color=white
--------------
0-9 -> 50
10-29 -> 60
30-49 -> 70
50-69 -> 80
70-89 -> 90
90-109 -> 100
110-129 -> 110
130-149 -> 120
150-169 -> 130
170-189 -> 140
190-209 -> 150
210-229 -> 160
230-249 -> 170
250-269 -> 180
270-289 -> 190
290-300 -> 200

Hero's Field base=100 color=green
--------------
0-4 -> 100
5-14 -> 110
15-24 -> 120
25-34 -> 130
35-44 -> 140
45-54 -> 150
55-64 -> 160
65-74 -> 170
75-84 -> 180
85-94 -> 190
95-104 -> 200
105-114 -> 210
115-124 -> 220
125-134 -> 230
135-144 -> 240
145-154 -> 250
155-164 -> 260
165-174 -> 270
175-184 -> 280
185-194 -> 290
195-204 -> 300
205-214 -> 310
215-224 -> 320
225-234 -> 330
235-244 -> 340
245-254 -> 350
255-264 -> 360
265-274 -> 370
275-284 -> 380
285-294 -> 390
295-300 -> 400

The Fungus Plains base=150 color=white
--------------
0-9 -> 50
10-29 -> 60
30-49 -> 70
50-69 -> 80
70-89 -> 90
90-109 -> 100
110-129 -> 110
130-149 -> 120
150-169 -> 130
170-189 -> 140
190-209 -> 150
210-229 -> 160
230-249 -> 170
250-269 -> 180
270-289 -> 190
290-300 -> 200

The Fungus Plains base=150 color=red
--------------
0-4 -> 100
5-14 -> 110
15-24 -> 120
25-34 -> 130
35-44 -> 140
45-54 -> 150
55-64 -> 160
65-74 -> 170
75-84 -> 180
85-94 -> 190
95-104 -> 200
105-114 -> 210
115-124 -> 220
125-134 -> 230
135-144 -> 240
145-154 -> 250
155-164 -> 260
165-174 -> 270
175-184 -> 280
185-194 -> 290
195-204 -> 300
205-214 -> 310
215-224 -> 320
225-234 -> 330
235-244 -> 340
245-254 -> 350
255-264 -> 360
265-274 -> 370
275-284 -> 380
285-294 -> 390
295-300 -> 400